plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -495,6 +495,7 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
                fc::copy( backup_dir / config::reversible_blocks_dir_name / "shared_memory.meta",
                          my->chain_config->blocks_dir / config::reversible_blocks_dir_name / "shared_memory.meta" );
             }
+          fc::remove_all( backup_dir ); // Replay complete, delete the unuseful backup_dir.
```

Comment:
- Replay is not complete at this point, so the comment is wrong.
- If `truncate-at-block` is specified that backup can be useful.
- If backups need to be deleted that can be done via scripts, not sure we want to remove backup_dir unless we know for sure it is no longer needed.

Reply:
- It is complete because functions are synchronous.
- ```truncate-at-block``` is only useful with --fix-reversible-blocks without a replay(**no ```backup_dir```**) or with --hard-replay-blockchain.
- Once the data is recovered, the old backup is useless.  Lots of backups lead to a complex data structure and space wasting.

Reply:
The actual replay doesn't happen until later.  What has happened by this point is `block_log::repair_log` and `recover_reversible_blocks`.  The comment is erroneous. 

I agree with @heifner  that the backups have value and can be cleaned up out-of-process.  That said, I would entertain a CLI option that prevents the backup from persisting after the process.

Reply:
I see.  But the CLI option is debatable because we can delete the backup through the OS.  Unless some validations need to be done before.